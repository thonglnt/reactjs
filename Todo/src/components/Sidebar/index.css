.sidebar-wrapper {
  width: 200px;
  height: 100%;
  background: var(--dark-charcoal);
  padding: 30px 0px;
  position: fixed;
}

@media screen and (max-width: 992px) {
  .sidebar-wrapper {
    width: 180px;
  }
}

@media screen and (max-width: 576px) {
  .sidebar-wrapper {
    width: 100px;
  }
}
