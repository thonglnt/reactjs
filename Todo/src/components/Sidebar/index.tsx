// Lib
import { ReactNode } from 'react';

export type SidebarProps = {
  children: ReactNode;
};

const Sidebar = ({ children }: SidebarProps) => {
  return <aside className='sidebar-wrapper'>{children}</aside>;
};

export default Sidebar;
