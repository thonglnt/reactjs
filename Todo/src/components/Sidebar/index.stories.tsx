// Lib
import { Story } from '@storybook/react';

// Components
import Sidebar, { SidebarProps } from '@components/Sidebar';

export default {
  title: 'Sidebar',
  component: Sidebar,
};

export const SidebarSample: Story<SidebarProps> = () => (
  <Sidebar children={<p>Sidebar component</p>} />
);
