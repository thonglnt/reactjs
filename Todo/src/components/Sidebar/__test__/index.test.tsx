// Lib
import { render, screen } from '@testing-library/react';

// Components
import Sidebar, { SidebarProps } from '@components/Sidebar';

describe('Test Sidebar component', () => {
  const initSidebar: SidebarProps = {
    children: <></>,
  };

  test('It should render correctly with initial Props', () => {
    const { container } = render(<Sidebar {...initSidebar} />);

    expect(container).toMatchSnapshot();
  });
});
