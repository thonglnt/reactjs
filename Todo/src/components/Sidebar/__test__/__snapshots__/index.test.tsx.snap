// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Sidebar component It should render correctly with initial Props 1`] = `
<div>
  <aside
    class="sidebar-wrapper"
  />
</div>
`;
